pkg/status/singletonstatus.go

```diff
@@ -18,69 +18,78 @@ package status
 
 import (
 	"context"
-	"fmt"
 
-	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
-	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
 	"github.com/kubestellar/kubestellar/pkg/util"
 )
 
-// reconcileSingletonByBdg goes over the binding-covered workload objects, decides sync or not for each of the workload objects,
-// then maintains their statuses based on the decisions.
 func (c *Controller) reconcileSingletonByBdg(ctx context.Context, bdgName string) error {
 	logger := klog.FromContext(ctx)
 	logger.V(4).Info("Reconciling singleton status due to binding change", "binding", bdgName)
 
-	binding, err := c.wdsKsClient.ControlV1alpha1().Bindings().Get(ctx, bdgName, metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// A deleted Binding is equivalent to one that references zero workload objects
-			return nil
+	wObjIDs, requested, nWECs := c.bindingPolicyResolver.GetSingletonReportedStateRequestsForBinding(bdgName)
+	for i := range wObjIDs {
+		wObjID, r, n := wObjIDs[i], requested[i], nWECs[i]
+		if !r || n != 1 {
+			if err := c.reconcileSingletonWObj(ctx, wObjID, false); err != nil {
+				return err
```

Comment:
I think it would be better if this error got wrapped with some explanation, mentioning the binding name, the fact that it arose in the source of syncing the WDS copies of status for that binding, and in particular trying to delete that status. You could do one wrapping here, or do part of the wrapping job in `c.syncBinding` (the stuff that is known at that point) and the remainder here (giving the info not known at the higher wrap site).

This is a generic code pattern issue.